stdlib/public/core/StringLegacy.swift

```diff
@@ -95,8 +95,33 @@ extension String {
   }
 }
 #else
-// FIXME: Implement hasPrefix and hasSuffix without objc
-// rdar://problem/18878343
+extension String {
+  /// Returns `true` iff `self` begins with `prefix`.
+  public func hasPrefix(prefix: String) -> Bool {
+    if prefix.isEmpty { return false }
```

Comment:
The goal here is to reproduce the NSString functionality. If you enter `"String".hasPrefix("")` in an Xcode Swift playground (where you're getting the Objective-C implementation at this time), you'll see that the result is `false`. `"String".hasPrefix("S")` is `true`, as expected.

In any event, the empty String as a parameter to `hasPrefix` is what is generally known as a trivial solution. Technically it is a substring of the operand, but that's not useful information.